src/sdk/config_fs.js

```diff
@@ -151,88 +218,184 @@ class ConfigFS {
      * @param {string} account_name
-     * @param {string} owner_root_account_id
+     * @param {string} owner_account_id
      * @returns {string} 
     */
-    get_iam_account_path_by_name(account_name, owner_root_account_id) {
-       // TODO -  change to this.symlink(account_name) on identities/ PR;
-       // update IAM user location identities/root_id_number/iam_account_name.symlink
-       return path.join(this.accounts_dir_path, this.json(account_name));
+    get_user_path_by_name(account_name, owner_account_id) {
+        // TODO - change to return path.join(this.identities_dir_path, owner_account_id, 'users', this.symlink(account_name));
+        return path.join(this.accounts_by_name_dir_path, this.symlink(account_name));
     }
 
     /**
-     * get_account_relative_path_by_name returns the full account path by name
+     * get_old_account_relative_path_by_name returns the old (5.17) full account path by name
      * @param {string} account_name
      * @returns {string} 
     */
-    get_account_relative_path_by_name(account_name) {
-       // TODO -  change to this.symlink(account_name) on identities/ PR;
-       return path.join('../', CONFIG_SUBDIRS.ACCOUNTS, this.json(account_name));
+    get_old_account_relative_path_by_name(account_name) {
+        return path.join('../', CONFIG_SUBDIRS.ACCOUNTS, this.json(account_name));
     }
 
     /**
-     * get_account_path_by_access_key returns the full account path by access key
+     * get_account_relative_path_by_id returns the full account path by id
+     * the target of symlinks will be the 
+     * @param {string} account_id
+     * @returns {string} 
+    */
+    get_account_relative_path_by_id(account_id) {
+       return path.join('../', CONFIG_SUBDIRS.IDENTITIES, account_id, this.json('identity'));
+    }
+
+    /**
+     * get_account_or_user_path_by_id returns the full account/user path by id as following
+     * {config_dir}/identities/{account_id}/identity.json
+     * @param {string} account_id
+     * @returns {string} 
+    */
+    get_identity_path_by_id(account_id) {
+        return path.join(this.identities_dir_path, account_id, this.json('identity'));
+    }
+
+    /**
+     * get_identity_by_id returns the full account/user data by id from the following path
+     * {config_dir}/identities/{account_id}/identity.json
+     * @param {string} id
+     * @param {string} [type]
+     * @param {{show_secrets?: boolean, decrypt_secret_key?: boolean, silent_if_missing?: boolean}} [options]
+     * @returns {Promise<Object>} 
+    */
+    async get_identity_by_id(id, type, options = {}) {
+        const account_path = this.get_identity_path_by_id(id);
+        let identity = await this.get_identity_config_data(account_path, { ...options, silent_if_missing: true });
+
+        if (!identity && type === TYPES.ACCOUNT) {
+            identity = await this.search_accounts_by_id(id, options);
```

Comment:
If we didn't find the identity by the ID, why do we read again in all the account configs by name and not just the old accounts?